acceptance/tests/ticket_3360_allow_duplicate_csr_with_option_set.rb

```diff
@@ -33,18 +41,19 @@
 
   new_certs = {}
   new_cert_list.stdout.each_line do |line|
-    if line =~ /^\+ (\S+) \((.+)\)$/
+    if line =~ /^\+ \"(\S+)\" \(?(.+)\)?$/ && agents_with_cert_name[$1]
       new_certs[$1] = $2
       puts "new cert: #{$1} #{$2}"
     end
   end
 
-  step "Verify the certs have changed"
+  assert_equal(agents.count, new_certs.count,
+               "Expected new number of agent csrs on master to equal number of agents")
+
   # using the agent name as the key may cause errors;
   # agent name from cfg file is likely to have short name
   # where certs might be signed with long names.
   old_certs.each_key { |key|
-    next if key.include? master # skip the masters cert, only care about agents
-    assert_not_equal(old_certs[key], new_certs[key], "Expected #{key} to have a changed key")
+    refute_equal(old_certs[key], new_certs[key], "Expected #{key} to have a changed key")
```

Comment:
> Why aren't we skipping the master's cert (or are you wanting to test the agent on the master)?

Yeah, I was wanting to test the agent on the master.  Seemed useful given that we often just run with that matrix in CI?

> Why is assert_not_equal not good?

I made a reference about that in the JIRA ticket.  I found that when running the beaker tests against a local VM that I'd get an exception, leading the test to fail, because `assert_not_equal` could not be found:

````
Test Case ./ruby/puppet/acceptance/tests/ticket_3360_allow_duplicate_csr_with_option_set.rb reported: #<RuntimeError: PuppetAcceptance::DSL::Helpers.with_puppet_running_on failed (check backtrace for location) because: undefined method `assert_not_equal' for #<Beaker::TestCase:0x007f97945fd218>
./ruby/puppet/acceptance/tests/ticket_3360_allow_duplicate_csr_with_option_set.rb:48:in `block (3 levels) in run_test'
````

It didn't look to me like `assert_not_equal` was defined in the minitest library, which I think is where we get most of these assertion helpers from, right?  So I'm not entirely sure why this hasn't blown up in our CI pipeline.  I was speculating that maybe because some other gems alias `assert_not_equal` to `refute_equal` that that maybe one of those was implicitly executed before those tests run in standard CI, not sure.  I was thinking `refute_equal` would be better since it is actually defined in the minitest that we use?

If you have more insight into what this problem may be, though, happy to go with your recommendation instead.